### PR TITLE
Add types to reraise util.

### PR DESCRIPTION
**Pull request by hramezani**
This patch adds types to `reraise` function.

### CODE REVIEW

File: src/urllib3/util/util.py [hunk 1]
```diff
@@ -25,7 +26,7 @@ def to_str(
     return x.decode()
 
 
-def reraise(tp, value, tb=None):
+def reraise(tp: type, value: Exception, tb: Optional[TracebackType] = None) -> NoReturn:
```
Comment (pquentin) [inline src/urllib3/util/util.py:29]: This function comes from six, and [typeshed uses the following types](https://github.com/python/typeshed/blob/0b1cd5989669544866213807afa833a88f649ee7/stubs/six/six/__init__.pyi#L93-L95):


```python3
def reraise(
    tp: Optional[Type[BaseException]], value: Optional[BaseException], tb: Optional[types.TracebackType] = ...
) -> NoReturn: ...
```

Maybe we should just copy that?

Comment (hramezani) [reply src/urllib3/util/util.py:29]: Thanks @pquentin for the suggestion. I will change it

**Review by pquentin — APPROVED** [review]

**Comment by pquentin** [comment]
The tests passed (https://github.com/urllib3/urllib3/actions/runs/571065039) but GitHub does not see it. Closing/reopening to try fixing this mess

**Comment by pquentin** [comment]
I don't understand why it says "waiting for status".

**Comment by sethmlarson** [comment]
I also don't understand, from the actions screen the PR is passing. I'm going to merge this. @hramezani have you updated your fork with the change to `master` -> `main`

**Comment by hramezani** [comment]
> @hramezani have you updated your fork with the change to `master` -> `main`

Yes, I did.

**Pull request merged**